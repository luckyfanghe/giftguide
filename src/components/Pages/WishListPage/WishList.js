import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Link, withRouter } from "react-router-dom";

import * as S from "./styled";
import banner1 from "../../../assets/images/banner1.webp";
import * as routes from "../../../constants/routes";

class WishList extends React.Component {
  constructor(props) {
    super(props);
  }

  _onAddClickHandler = event => {
    event.preventDefault();
    this.props.history.push(routes.ADD_WISH_ITEM_PAGE);
  };

  render() {
    return (
      <React.Fragment>
        <S.Body data-testid="WishList">
          <S.Title>WishList</S.Title>
          <S.ShareButton type="button" border>
            Share Items
          </S.ShareButton>
          <S.WishListContainer>
            <S.WishItem>
              <S.AddWishItem border onClick={this._onAddClickHandler}>
                <S.Plus border>+</S.Plus>
              </S.AddWishItem>
              <S.WishItemLabel>add to wish list</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.WishItemImage src={banner1} />
              <S.WishItemLabel>Wish Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
          </S.WishListContainer>
          <S.ReceivedItemsLabel>RECEIVED ITEMS</S.ReceivedItemsLabel>
          <S.WishListContainer>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Received </S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Received </S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
            <S.WishItem>
              <S.ReceivedItemImage src={banner1} />
              <S.WishItemLabel>Received Item Name Item Name</S.WishItemLabel>
              <S.WishItemLabel>$ 67</S.WishItemLabel>
            </S.WishItem>
          </S.WishListContainer>
        </S.Body>
      </React.Fragment>
    );
  }
}

const actions = dispatch => bindActionCreators({}, dispatch);

function mapStateToProps(state) {
  return {
    user: state.userReducer.user,
    userState: state.userReducer.userState
  };
}

export default connect(mapStateToProps, actions)(withRouter(WishList));
