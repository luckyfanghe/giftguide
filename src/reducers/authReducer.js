// @flow
import types from "../actions/types";

const initialState = {
  sidebarOpened: true,
  sidebarStyle: "login",
  authState: ""
};
function authReducer(state = initialState, action) {
  switch (action.type) {
    case types.AUTH.OPENSIDEBAR:
      return {
        ...state,
        sidebarOpened: true,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.CLOSESIDEBAR:
      return {
        ...state,
        sidebarOpened: false,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.DOINGAUTH:
      return {
        ...state,
        authState: action.state
      };
    default:
      return {
        ...state
      };
  }
}

export default authReducer;
