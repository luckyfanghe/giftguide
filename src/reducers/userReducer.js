// @flow
import types from "../actions/types";

const initialState = {
  user: null,
  userState: "",
  userStateDes: ""
};
function userReducer(state = initialState, action) {
  switch (action.type) {
    case types.USER.UPDATE:
      return {
        ...state,
        user: action.userInfo
      };
    case types.USER.LOGINFAILED:
      return {
        ...state,
        userState: "login failed",
        userStateDes: action.state
      };
    case types.USER.LOGINSUCCESS:
      return {
        ...state,
        userState: "login success",
        userStateDes: ""
      };
    case types.USER.SIGNUPFAILED:
      return {
        ...state,
        userState: "signup failed",
        userStateDes: action.state
      };
    case types.USER.SIGNUPSUCCESS:
      return {
        ...state,
        userState: "signup success",
        userStateDes: ""
      };
    default:
      return {
        ...state
      };
  }
}

export default userReducer;
