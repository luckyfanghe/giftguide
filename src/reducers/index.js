import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";

import userReducer from "./userReducer";
import authReducer from "./authReducer";

export default combineReducers({
  userReducer,
  authReducer
});
