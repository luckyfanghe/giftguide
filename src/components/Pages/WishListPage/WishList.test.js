import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import reducers from "../../../reducers";
import rootSaga from "../../../sagas";
import WishList from "./WishList";
import { BrowserRouter } from "react-router-dom";

const sagaMiddleware = createSagaMiddleware();
const enhancers = composeWithDevTools(applyMiddleware(sagaMiddleware));

const store = createStore(reducers, enhancers);

sagaMiddleware.run(rootSaga);

it("WishList renders corrently", () => {
  const { queryByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
        <WishList />
      </BrowserRouter>
    </Provider>
  );

  expect(queryByTestId("WishList")).toBeTruthy();
});

describe("WishList", () => {
  test("WishList snapshot renders ", () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <WishList />
        </BrowserRouter>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
