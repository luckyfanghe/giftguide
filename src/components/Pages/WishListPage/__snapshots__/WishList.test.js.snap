// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishList WishList snapshot renders  1`] = `
<div
  className="sc-AykKC hdBJOi"
  data-testid="WishList"
>
  <div
    className="sc-AykKD dkLhfl"
  >
    WishList
  </div>
  <button
    className="sc-AykKE gYQywC"
    type="button"
  >
    Share Items
  </button>
  <div
    className="sc-AykKF gvLVwl"
  >
    <div
      className="sc-AykKG jyvbok"
    >
      <div
        className="sc-AykKH uhKLd"
        onClick={[Function]}
      >
        <div
          className="sc-AykKK kobeoF"
        >
          +
        </div>
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        add to wish list
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKI cYiOOi"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Wish Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
  </div>
  <div
    className="sc-fzXfLP gnDrIA"
  >
    RECEIVED ITEMS
  </div>
  <div
    className="sc-AykKF gvLVwl"
  >
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Received 
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Received 
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
    <div
      className="sc-AykKG jyvbok"
    >
      <img
        alt="banner"
        className="sc-AykKJ hSxWEz"
        src="banner1.webp"
      />
      <div
        className="sc-fzXfLO lgtSrV"
      >
        Received Item Name Item Name
      </div>
      <div
        className="sc-fzXfLO lgtSrV"
      >
        $ 67
      </div>
    </div>
  </div>
</div>
`;
